src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -120,14 +124,16 @@ public Person toModelType() throws IllegalValueException {
         }
         final Description modelDescription = new Description(description);
 
-        if (nextOfKin == null) {
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
-                    NextOfKin.class.getSimpleName()));
+        final Optional<NextOfKin> modelNextOfKin;
+        // If NOK field was not filled
+        if (nextOfKin == null || nextOfKin.equals("null")) {
+            modelNextOfKin = Optional.empty();
```

Comment:
You can also use `map(nok -> {if (!NextOfKin.isValidNextOfKin(nok) {throw new ...} else {return nok})`

Reply:
during testing, you should check if the "null" string gets returned at any point, the annotation `@Include...` should prevent the "null" string from being passed through. You can directly check the addressbook.json file as well

Reply:
> `modelNextOfKin = Optional.ofNullable(nextOfKin).filter(NextOfKin::isValidNextOfKin);`
> 
> There is no need for the exception anymore right because we can hold empty values now. But if you want to still throw the exception, you can filter by not valid, and then have a `ifPresent(nok -> throw new IllegalValue...)` consumer that will throw an exception

I think its important to throw the Exception for Illegal Values that are not empty. For example, things like "James$$%%".

Will try using a shorter way to represent the if statements though.

Reply:
> You can also use `map(nok -> {if (!NextOfKin.isValidNextOfKin(nok) {throw new ...} else {return nok})`

This is an interesting idea, but we would essentially be throwing the exception inside the optional instead of up to the function that handles it.
<img width="720" alt="image" src="https://github.com/AY2324S2-CS2103T-W12-4/tp/assets/48405651/1c1bc6bf-25b5-48fa-916f-4229a0f1d706">
Something like this^.
So I would have to catch the exception myself, as opposed to how it is supposed to be handled.

Reply:
ah because the exception is a checked exception, my bad.

something like this would suffice maybe:

`if (nextOfKin != null && !NextOfKin.isValidNextOfKin(nextOfKin)) {
    throw new Illegal...
} 

modelNextOfKin = Optional.ofNullable(nextOfKin).map(NextOfKin:new);